From 10106fdb4cc833faa3c2fcb980d0a4eb121fe19d Mon Sep 17 00:00:00 2001
From: Fabrice Fontaine <fontaine.fabrice@gmail.com>
Date: Tue, 23 Jul 2024 10:31:32 +0200
Subject: [PATCH] wminput/plugins/led/led.c: fix build with gcc >= 14

Fix the following build failure with gcc >= 14:

led.c: In function 'wmplugin_exec':
led.c:86:42: error: assignment to 'struct cwiid_btn_message *' from incompatible pointer type 'struct cwiid_btn_mesg *' [-Wincompatible-pointer-types]
   86 |                                 btn_mesg = &mesg[i].btn_mesg;
      |                                          ^

Fixes:
 - http://autobuild.buildroot.org/results/a3bde74ff2137d088f4261e62930859bfe460cb9

Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>
Upstream: https://github.com/abstrakraft/cwiid/pull/47
---
 wminput/plugins/led/led.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/wminput/plugins/led/led.c b/wminput/plugins/led/led.c
index 93b96d9..591f365 100644
--- a/wminput/plugins/led/led.c
+++ b/wminput/plugins/led/led.c
@@ -72,7 +72,7 @@ struct wmplugin_data *wmplugin_exec(int mesg_count, union cwiid_mesg mesg[])
 {
 	int i;
 	uint8_t button;
-	struct cwiid_btn_message *btn_mesg;
+	struct cwiid_btn_mesg *btn_mesg;
 
 	uint8_t led_state = (Led1 ? CWIID_LED1_ON : 0)
 	                  | (Led2 ? CWIID_LED2_ON : 0)
-- 
2.43.0

